src/pkg/repository/repository.go

```diff
@@ -272,6 +273,42 @@ func (r *repository) Connect(ctx context.Context) (err error) {
 	return nil
 }
 
+// UpdatePassword updates Kopia password
+func (r *repository) UpdatePassword(ctx context.Context, password string) (err error) {
+	ctx = clues.Add(
+		ctx,
+		"acct_provider", r.Account.Provider.String(),
+		"acct_id", clues.Hide(r.Account.ID()),
+		"storage_provider", r.Storage.Provider.String())
+
+	defer func() {
+		if crErr := crash.Recovery(ctx, recover(), "repo connect"); crErr != nil {
+			err = crErr
+		}
+	}()
+
+	progressBar := observe.MessageWithCompletion(ctx, "Connecting to repository")
+	defer close(progressBar)
+
+	kopiaRef := kopia.NewConn(r.Storage)
+	if err := kopiaRef.Connect(ctx, r.Opts.Repo); err != nil {
+		return clues.Wrap(err, "connecting kopia client")
+	}
+
+	defer kopiaRef.Close(ctx)
+
+	repository := kopiaRef.Repository.(repo.DirectRepository)
+	err = repository.FormatManager().ChangePassword(ctx, password)
```

Comment:
This doesn't feel like the correct owner for this code.  It surfaces too much of kopia internals in a package that strives to avoid details like that.  It would be better for the kopia package to own this behavior.  Further, it would be best to have that functionality abide some standardized interface so that this can be tested with mocks instead of only using integration testing.